### ISSUE TITLE
upgrade to guava 20 cause bunch of violations in code by IntelijIdea

### ISSUE THREAD

**Issue by romani** [google/guava#2629]
here is example of how our project tried to upgrade to guava 20 from 19 (simple version bump) - https://github.com/checkstyle/checkstyle/pull/3527.

it result in violations by IntelijIdea:
![screenshot- upgradetoguava20](https://cloud.githubusercontent.com/assets/812984/20042839/f168ad4a-a435-11e6-9ed8-d08c6b8a8778.png)
https://teamcity.jetbrains.com/viewLog.html?buildId=917921&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest

All is that only due to commit: https://github.com/google/guava/commit/f6da31136af15dd324d56d253bc17c5b9afe9759#diff-7ed40f507101a8625df970c98abcbc55

It is used jdk class `javax.annotation.CheckReturnValue` to trigger validation and google custom class `com.google.errorprone.annotations.CanIgnoreReturnValue` to suppress violation.

yes it works for google, but now for others. I think it is not accurate to use different markers for triggering and suppressions. If you want to use errorprones's `CanIgnoreReturnValue` please use smth like `com.google.errorprone.annotations.CheckReturnValue` (non existing class).

but usage of jdk marker/annotation trigger validation process of other tools. Extra dependencies are also not desirable.

It would be better to use `CheckReturnValue` on certain methods as show at - http://errorprone.info/bugpattern/CheckReturnValue.
or to use some friendly(common for all) approaches that other tools can use and ignore that methods.

**Comment by cpovirk** [comment]
Having methods be `@CheckReturnValue` by default is important to us, so we don't want to have to put it on each method individually. The IntelliJ fallout is pretty sad, though :( One thing that would be good to do is to ask the IntelliJ people if they can respect `@CanIgnoreReturnValue`. And as you note, it may be worth [asking the Error Prone team](https://github.com/google/error-prone/issues/new) if they would add their own `@CheckReturnValue` annotation. (And if you haven't already checked, it would be good to verify that IntelliJ recognizes only `javax.annotation.CheckReturnValue` and not any other annotation with simple name `CheckReturnValue`.)

**Comment by romani** [comment]
> by default is important to us,

I hope you understand that guava is library so "us" should be "us and users". In other case popularity of guava will decrease. CIs will not let to upgrade to new version of guava.
I just gave example in simple opensource project.

I posted an note at error-prone team - https://github.com/google/error-prone/issues/475

**Comment by kartoffelsup** [comment]
You are quite demanding for someone who is using great software free of charge. :)

**Comment by kevinb9n** [comment]
Nice of you to say, kartoffelsup, thanks.

I understand romani's point, though. It's just that I think we can solve this so that both sides are happy; it's premature to talk about an us vs. our users conflict.

**Comment by romani** [comment]
I just want to stress the point that solution should be consistent. It either jdk's annotations or errorprone's . Either way will satisfy community who use other static analysis tools in CIs.

**Comment by michaelhixson** [comment]
Umm, I may deserve some of the blame here.  I made a feature request for IntelliJ to support `@CheckReturnValue` on classes and packages (previously they supported it on methods only), and they added that feature, but I neglected to ask for `@CanIgnoreReturnValue` support at the same time.  I eventually did ask for `@CanIgnoreReturnValue` support and supposedly that is coming in the next minor version of IntelliJ, 2016.3.  I was hoping that would be released before Guava 20, but sadly it did not work out that way.  Sorry!

See https://youtrack.jetbrains.com/issue/IDEA-158576
